pages/app/models/refinery/page.rb

```diff
@@ -73,6 +73,9 @@ def should_generate_new_friendly_id?
     before_destroy :deletable?
     after_save :reposition_parts!
 
+    after_save { update_all_descendants if descendants.any? }
+    after_move { update_all_descendants if descendants.any? }
```

Comment:
I don't think you need the conditional, and possibly you can just use:

```ruby
after_save :update_all_descendants
after_move :update_all_descendants
```

Reply:
You're right